### PR TITLE
improve integration tests

### PR DESCRIPTION
**Pull request by vkagamlyk**
*Issue #, if available:*
https://bitquill.atlassian.net/browse/AN-1100

*Description of changes:*
improve integration tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docker/gremlin-server/gremlin-server-integration.yaml [hunk 1]
```diff
@@ -24,8 +24,7 @@ graphs: {
   modern: conf/tinkergraph-empty.properties,
   crew: conf/tinkergraph-empty.properties,
   grateful: conf/tinkergraph-empty.properties,
-  sink: conf/tinkergraph-empty.properties
-}
+  sink: conf/tinkergraph-empty.properties}
```
Comment (lyndonbauto) [inline docker/gremlin-server/gremlin-server-integration.yaml]: Was this autoformat?

Comment (vkagamlyk) [reply docker/gremlin-server/gremlin-server-integration.yaml]: no, I tried to change something and then made like in other configs: https://github.com/lyndonbauto/tinkerpop/blob/master/docker/gremlin-server/gremlin-server-integration-secure.yaml#L27

Comment (vkagamlyk) [reply docker/gremlin-server/gremlin-server-integration.yaml]: reverted to reduce noise

**Comment by xiazcy** [comment]
As per Stephen's comment on MS5 PR: https://github.com/apache/tinkerpop/pull/1619#discussion_r851227771, there is the concern of duplicated configuration files that are hard to maintain. 

I had approached this Docker set-up as an intermediate solution for getting MS5 integration tests to run on GitHub Actions, and so avoided changing codes outside of gremlin-go package, but in light of Stephen's comment, it may be best to use the same files. This may either involve modifying the original scripts outside of the gremlin-go package, or just change our configuration around using them.

I know Valentyn's original intention here was to modify the original set up files, but then had difficulties with failing tests in other GLVs due to the changes, and so I suggested to isolate the file changes to the gremlin-go Docker folder. While the tests pass now, it is probably not a sustainable solution. Perhaps we can have a quick discussion about this tomorrow after standup?

**Comment by lyndonbauto** [comment]
> Perhaps we can have a quick discussion about this tomorrow after standup?

Sounds good. Thanks for the insight @xiazcy.

File: gremlin-go/driver/connection_test.go [hunk 2]
```diff
@@ -771,10 +752,7 @@ func TestConnection(t *testing.T) {
 		readUsingAnonymousTraversal(t, g)
 
 		// Drop the graph and check that it is empty.
-		dropGraph(t, g)
-		readCount(t, g, "", 0)
-		readCount(t, g, testLabel, 0)
-		readCount(t, g, personLabel, 0)
+		resetGraph(t, g)
```
Comment (lyndonbauto) [inline gremlin-go/driver/connection_test.go:755]: Does resetting the graph defeat the race condition? I thought we'd still have the same issue where it is concurrently writing to the graph. Perhaps I am missing something.

Comment (xiazcy) [reply gremlin-go/driver/connection_test.go:755]: I believe Valentyn added a new graph `gtest` specifically for the connection tests, so they'd be accessing separate graphs from the cucumber suite.

Comment (vkagamlyk) [reply gremlin-go/driver/connection_test.go:755]: `resetGraph` does not remove race condition. 
As I understood we have conflict with execution of connection and cucumber test in parallel, so I introduced gtest to not use `g` or `ggrpaph`. But I can be wrong :)

@xiazcy 👍

Comment (lyndonbauto) [reply gremlin-go/driver/connection_test.go:755]: Ack - I thought it was due to concurrent tests here running. Nice fix. Approved.

Comment (lyndonbauto) [reply gremlin-go/driver/connection_test.go:755]: Or actually, I will wait till our conversation tomorrow to approve.

**Review by lyndonbauto — DISMISSED** [review]

**Review by lyndonbauto — APPROVED** [review]

**Review by xiazcy — APPROVED** [review]

**Pull request merged**